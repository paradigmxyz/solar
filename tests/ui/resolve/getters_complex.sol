contract Complex {
    struct A {
        B b;
    }

    struct B {
        uint256[] arr;
    }

    mapping(uint256 => A) public mapA;

    function pushValueA(uint256 idx, uint256 val) public {
        mapA[idx].b.arr.push(val);
    }

    mapping(uint256 => B) public mapB; //~ ERROR: getter must return at least one value

    function pushValueB(uint256 idx, uint256 val) public {
        mapB[idx].arr.push(val);
    }
}
