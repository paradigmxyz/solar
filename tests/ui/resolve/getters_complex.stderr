error: getter must return at least one value
  --> ROOT/tests/ui/resolve/getters_complex.sol:LL:CC
   |
LL |     mapping(uint256 => B) public mapB;
   |     ^^^^^^^^^^^^^^^^^^^-^^^^^^^^^^^^^^
   |                        |
   |                        note: the struct has all its members omitted, therefore the getter cannot return any values
   |

error: aborting due to 1 previous error

